models/world_coordinates.py

```diff
@@ -152,4 +120,72 @@ def get_downstreams(self, session=None):
 
         downstreams = zps + subs
         return downstreams
+
+    def save( self, filename=None, **kwargs ):
+        """Write the WCS data to disk.
+        Updates self.filepath
+        Parameters
+        ----------
+          filename: str or path
+             The path to the file to write, relative to the local store
+             root.  Do not include the extension (e.g. '.psf') at the
+             end of the name; that will be added automatically.
+             If None, will call image.invent_filepath() to get a
+             filestore-standard filename and directory.
+          Additional arguments are passed on to FileOnDiskMixin.save
+        """ 
+
+        # ----- Make sure we have a path ----- #
+        # if filename already exists, check it is correct and use
+
+        if filename is not None:
+            if not filename.endswith('.txt'):
+                filename += '.txt'
+            self.filepath = filename
+
+        # if not, generate one
+        else:
+            if self.image.filepath is not None:
+                self.filepath = self.image.filepath
+            else:
+                self.filepath = self.image.invent_filepath()
+
+            if self.provenance is None:
+                raise RuntimeError("Can't invent a filepath for the WCS without a provenance")
```

Comment:
Maybe move this check above the invent filepath so it fails without ever making a partial filename